src/main/java/core/basesyntax/service/amount/PurchaseHandlerImpl.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.service.amount;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandlerImpl implements OperationHandler {
+    @Override
+    public void process(FruitTransaction fruitTransaction) {
+        Storage.fruits.put(fruitTransaction.getFruitType(),
+                (Storage.fruits.get(fruitTransaction.getFruitType())
+                        - fruitTransaction.getAmount()));
```

Comment:
```suggestion
       Storage.fruits.compute(fruitTransaction.getFruitType(), (k, v) -> v - fruitTransaction.getAmount());
```
same for other operations